docs/blog/posts/ebpf-container-challenges.md

```diff
@@ -0,0 +1,509 @@
+lrwxrwxrwx 1 root root 0 Feb 15 12:10 time -> 'time:[4026531834]'
+lrwxrwxrwx 1 root root 0 Feb 15 12:10 time_for_children -> 'time:[4026531834]'
+lrwxrwxrwx 1 root root 0 Feb 15 12:10 user -> 'user:[4026531837]'
+lrwxrwxrwx 1 root root 0 Feb 15 12:10 uts -> 'uts:[4026532857]'
+```
+
+In this case, the mount namespace file is `/proc/102745/ns/mnt`. How to find the
+PID is discussed below.
+
+### Finding Your Way Back Home From a Different Namespace
+
+In the Kubernetes implementation, `/proc` is mounted in the `bpfman-agent`
+container so it has access to the ns directories of other containers. However,
+after the `bpfman-agent` does a `setns` to the target, it loses access to the
+`/proc` mount. Therefore, the `bpfman-agent` also opens it's own mount namespace
+file prior to calling `setns` so it can get back to its own namespace after
+attaching the uprobe in the target container.
+
+### Running `setns` From a Multi-threaded Process
+
+Calling `setns` to a mount namespace doesn't work from a multi-threaded process.
+
+The solution chosen was to create a standalone single-threaded executable called
+[bpfman-ns](https://github.com/bpfman/bpfman/blob/main/bpfman-ns/src/main.rs)
+that does the job of entering the namespace, attaching the uprobe, and then
+returning to the bpfman namespace to save the needed info.
+
+### Finding the PID
+
+#### Finding a Host Container PID on a Linux Server
+
+This section provides an overview of PID namespaces and shows several ways to
+find the host PID for a container.
+
+##### tl;dr
+
+If you used Podman or Docker to run your container, **and** you gave the
+container a unique name, the following commands can be used to find the host PID
+of a container.
+
+```bash
+podman inspect -f '{{.State.Pid}}' <CONTAINER_NAME>
+```
+
+or, similarly,
+
+```bash
+docker inspect -f '{{.State.Pid}}'  <CONTAINER_NAME>
+```
+
+##### Overview of PID namespaces and Container Host PIDs 
+
+Each container has a PID namespace. Each PID namespace (other than the root) is
+contained within a parent PID namespace. In general, this relationship is
+hierarchical and PID namespaces can be nested within other PID namespaces. In
+this section, we will just cover the case of a root PID namepsace on a Linux
+server that has containers with PID namespaces that are direct children of the
+root. The multi-level case is described in the section on Nested Containers with
+kind below.
+
+The PID namespaces can be listed using the `lsns -t pid` command. Before we
+start any containers, we just have the one root pid namespace as shown below.
+
+```bash
+sudo lsns -t pid
+        NS TYPE NPROCS PID USER COMMAND
+4026531836 pid     325   1 root /usr/lib/systemd/systemd rhgb --switched-root --system --deserialize 30
+```
+
+Now lets start a container with the following command in a new shell:
+
+```bash
+podman run -it --name=container_1 fedora:latest /bin/bash
+```
+
+>> **NOTE:** In this section, we are using `podman` to run containers. However,
```

Comment:
```suggestion
> **NOTE:** In this section, we are using `podman` to run containers. However,
```
Probably only need the single indention here.

Reply:
Maybe, but I use a plugin called Rewrap that inserts the second one and it also works.